internal/transport/http2_client.go

```diff
@@ -223,6 +223,9 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 	// Any further errors will close the underlying connection
 	defer func(conn net.Conn) {
 		if err != nil {
+			if logger.V(logLevel) {
+				logger.Errorf("transport: closing connection due to error %v.", err)
```

Comment:
Sgtm to put information that this happened in newHTTP2Client. Is the information that it will Close connection appropriate information to put into an error message? I see what you're saying you can return the information in the error instead.